api/src/main/java/org/pmiops/workbench/auth/DelegatedUserCredentials.java

```diff
@@ -0,0 +1,165 @@
+ * to authorize calls as end users without loading private keys directly into the application.
+ *
+ * <p>This class shares some patterns in common with the ImpersonatedCredentials class; namely, it
+ * uses the IAM Credentials API to allow one service account to perform some actions on behalf of
+ * another service account. However, this class differs in two notable ways: (1) it supports
+ * impersonation of end users, while ImpersonatedCredentials supports only impersonation of service
+ * accounts, and (2) it relies on application default credentials for simplicity in the All of Us
+ * Researcher Workbench use case.
+ *
+ * <p>Example usage, for authorizing user requests to the Google Directory API:<br>
+ *
+ * <pre>
+ *   DelegatedUserCredentials delegatedCredentials = new DelegatedUserCredentials(
+ *     "service-account-with-dwd-enabled@project-name.iam.gserviceaccount.com",
+ *     "admin-gsuite-user@my-gsuite-domain.com",
+ *     DirectoryScopes.ADMIN_DIRECTORY_USERS);
+ *   Directory directoryClient = new Directory.Builder(
+ *       new NetHttpTransport(), new JacksonFactory(), null)
+ *       .setHttpRequestInitializer(new HttpCredentialsAdapter(delegatedCredentials))
+ *       .build();
+ * </pre>
+ */
+public class DelegatedUserCredentials extends OAuth2Credentials {
+
+  static final String JWT_BEARER_GRANT_TYPE = "urn:ietf:params:oauth:grant-type:jwt-bearer";
+  static final String SERVICE_ACCOUNT_NAME_FORMAT = "projects/-/serviceAccounts/%s";
+  static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+  public static final Duration ACCESS_TOKEN_DURATION = Duration.ofMinutes(60);
+
+  // The email of the service account whose system-managed key should be used to sign the JWT
+  // assertion which is exchanged for an access token. This service account:
+  // - Must have domain-wide delegation enabled for the target user's G Suite domain and scopes.
+  // - Does not need to be the same service account (SA) as the application default credentials
+  // (ADC)
+  //     service account. If they are different, the ADC account must have the Service Account Token
+  //     Creator role granted on this service account. See
+  //     https://cloud.google.com/iam/docs/creating-short-lived-service-account-credentials for more
+  //     details.
+  private String serviceAccountEmail;
+  // The full G Suite email address of the user for whom an access token will be generated.
+  private String userEmail;
+  // The set of Google OAuth scopes to be requested.
+  private List<String> scopes;
+  // The HttpTransport to be used for making requests to Google's OAuth2 token server. If null,
+  // a default NetHttpTransport instance is used.
+  private HttpTransport httpTransport;
+  // The IAM Credentials API client to be used for fetching a signed JWT from Google. If null,
+  // a default API client will be used.
+  private IamCredentialsClient client;
```

Comment:
The documentation for this client indicates that it should be closed: https://github.com/googleapis/java-iamcredentials/blob/master/google-cloud-iamcredentials/src/main/java/com/google/cloud/iam/credentials/v1/IamCredentialsClient.java#L56-L57

Providing this as a spring singleton and moving this to be a constructor parameter (callers would inject the client and pass it through) would probably be my preferred solution to this.